frontend/src/components/forms/CreateLog.tsx

```diff
@@ -119,7 +119,7 @@ const CreateLog = ({ getRecords, countRecords, setUserPageNum }: Props) => {
       hour12: false,
     }),
   );
-  const [building, setBuilding] = useState("");
+  const [buildingId, setBuildingId] = useState<number | null>(null);
```

Comment:
The default here can be just set to -1 like what's done below for `resident` and `attnTo`